crates/compiler/mono/src/ir.rs

```diff
@@ -4268,35 +4268,50 @@ pub fn with_hole<'a>(
     let arena = env.arena;
 
     match can_expr {
-        Int(_, _, int_str, int, _bound) => assign_num_literal_expr(
-            env,
-            layout_cache,
-            assigned,
-            variable,
-            &int_str,
-            IntOrFloatValue::Int(int),
-            hole,
-        ),
+        Int(_, _, int_str, int, _bound) => {
+            match assign_num_literal_expr(
+                env,
+                layout_cache,
+                assigned,
+                variable,
+                &int_str,
+                IntOrFloatValue::Int(int),
+                hole,
+            ) {
+                Ok(stmt) => stmt,
+                Err(_) => hole.clone(),
```

Comment:
I could not figure out how but hole here has an erroneous stmt when there's an error. Using hole, the error message would be the same as the one from using `Str`.

Str
```
» List.dropAt [1, 2, 3] "abc"
This Roc code crashed with: "Hit an erroneous type when creating a layout for `List.dropAt`"

── TYPE MISMATCH ───────────────────────────────────────────────────────────────

This 2nd argument to dropAt has an unexpected type:

4│      List.dropAt [1, 2, 3] "abc"
                              ^^^^^

The argument is a string of type:

    Str

But dropAt needs its 2nd argument to be:

    Nat

```

negative number

```
» List.dropAt [1, 2, 3] -3
This Roc code crashed with: "Hit an erroneous type when creating a layout for `List.dropAt`"

── TYPE MISMATCH ───────────────────────────────────────────────────────────────

This 2nd argument to dropAt has an unexpected type:

4│      List.dropAt [1, 2, 3] -3
                              ^^

The argument is a number of type:

    I8, I16, F32, I32, F64, I64, I128, or Dec

But dropAt needs its 2nd argument to be:

    Nat

```